fe/src/main/java/org/apache/doris/analysis/CreateRoutineLoadStmt.java

```diff
@@ -394,6 +400,17 @@ private void checkKafkaProperties() throws AnalysisException {
                 }
             }
         }
+        // check custom kafka property
+        for (Map.Entry<String, String> dataSourceProperty : dataSourceProperties.entrySet()) {
+            if (dataSourceProperty.getKey().startsWith("property.")) {
+                String propertyValueArr[] = dataSourceProperty.getKey().split("\\.");
+                if (propertyValueArr.length < 2) {
+                    throw new AnalysisException("kafka property value could not be a empty string");
+                }
+                customKafkaProperties.put(dataSourceProperty.getKey(), dataSourceProperty.getValue());
```

Comment:
use the key which is removed prefix "property.” from dataSourceProperty.getKey()